src/Eloquent/EloquentDeferredValue.php

```diff
@@ -26,12 +27,12 @@ public static function resolve(array $deferredValues): void
     {
         static::loadEloquentRelations($deferredValues);
 
-        collect($deferredValues)
```

Comment:
Is `collect()` deprecated?

Reply:
It's not and is still available. I thought that it would be nice not to depend on a global function included from the Laravel helpers file, so I switched to using Collection class directly. However it was a bit of a 'half assed' attempt because there are still calls to data_get (which is only available as such global helper, and not a class I can hook into). So all in all, not a massive improvement and probably a bit of a useless change :D

Reply:
Does `Illuminate\Support\Arr::get()` not cut the mustard?

Reply:
Hard to say. Arr::get passes the tests, but looking at the implementation of both, I am not sure if Arr::get covers all the use cases that data_get does. The 'relation' string is controlled by the users of this package, so I can't be sure that it never uses e.g. '*' which is valid segment for data_get but not for Arr:get. In theory it shouldn't be, and the only supported 'relation' strings should be compatible with the Laravel Eloquent relation syntax. I think this change would require a new major version of this package just in case, and then I would also like to look at improved handling of the relation string in general - so would prefer to leave it as is, and then do the switch to Arr::get as part of this work. However, thanks for throwing that excellent suggestion, as I would definitely want to switch to that in the future! :D

Reply:
Agreed, they do look a little different.